include/phasar/PhasarLLVM/AnalysisStrategy/WholeProgramAnalysis.h

```diff
@@ -44,7 +44,8 @@ class WholeProgramAnalysis {
   std::unique_ptr<PointerAnalysisTy> PointerInfo;
   std::unique_ptr<CallGraphAnalysisTy> CallGraph;
   std::set<std::string> EntryPoints;
-  std::unique_ptr<ConfigurationTy> Config;
+  ConfigurationTy *Config = nullptr;
+  bool OwnsConfig = false;
```

Comment:
This is, because the `Config` is not always owned. There are two constructors that initialize the Config:
- One that creates a new one - this should be owned
- One that gets passed as parameter and should not be owned - in fact, most of the time it will be stack allocated

Therefore, using a `unique_ptr` does not work well here.